hilti/toolchain/include/compiler/global-optimizer.h

```diff
@@ -27,7 +28,9 @@ struct GlobalOptimizer {
         bool referenced = false;
     };
 
-    using Functions = std::map<std::tuple<ModuleID, StructID, FieldID>, Uses>;
+    enum class Type { Method, Function };
+    using ID = std::pair<Type, std::string>;
```

Comment:
Could the `std:string` be an `ID`?

Reply:
Yes. The original form in this patch was for some larger set of changes to also track types; I'll add that back if and once needed.